bentoml/_internal/frameworks/transformers.py

```diff
@@ -305,39 +354,42 @@ def save_model(
         )
 
     if task_name is not None and task_definition is not None:
-        from transformers.pipelines import SUPPORTED_TASKS
-
         try:
-            import cloudpickle  # type: ignore
-        except ImportError:  # pragma: no cover
-            raise MissingDependencyException(
-                "Module `cloudpickle` is required in order to use to save custom pipelines."
-            )
+            from transformers.pipelines import PIPELINE_REGISTRY
 
-        logger.info(
-            f"Arguments `task_name` and `task_definition` are provided. Saving model with pipeline "
-            f"task name '{task_name}' and task definition '{task_definition}'."
-        )
+            PIPELINE_REGISTRY.register_pipeline(task_name, task_definition)
+        except ImportError:
+            from transformers.pipelines import TASK_ALIASES
+            from transformers.pipelines import SUPPORTED_TASKS
 
-        if pipeline.task is None or pipeline.task != task_name:
-            raise BentoMLException(
-                f"Argument `task_name` '{task_name}' does not match pipeline task name '{pipeline.task}'."
+            logger.info(
+                f"Arguments `task_name` and `task_definition` are provided. Saving model with pipeline "
+                f"task name '{task_name}' and task definition '{task_definition}'."
             )
 
-        impl: type = task_definition["impl"]
-        if type(pipeline) != impl:
-            raise BentoMLException(
-                f"Argument `pipeline` is not an instance of {impl}. It is an instance of {type(pipeline)}."
-            )
+            if pipeline.task is None or pipeline.task != task_name:
+                raise BentoMLException(
+                    f"Argument `task_name` '{task_name}' does not match pipeline task name '{pipeline.task}'."
+                )
 
-        if task_name in SUPPORTED_TASKS:
-            if SUPPORTED_TASKS[task_name] != task_definition:
+            impl: type = task_definition["impl"]
+            if type(pipeline) != impl:
                 raise BentoMLException(
-                    f"Argument `task_definition` '{task_definition}' does not match pipeline task "
-                    "definition '{SUPPORTED_TASKS[task_name]}'."
+                    f"Argument `pipeline` is not an instance of {impl}. It is an instance of {type(pipeline)}."
                 )
-        else:
-            SUPPORTED_TASKS[task_name] = task_definition
+            if task_name in TASK_ALIASES:
```

Comment:
if task is an alias, then we should throw an error warning users not to override them.